OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/preferencecenter/OptimovePreferenceCenter.java

```diff
@@ -0,0 +1,290 @@
+import com.optimove.android.main.common.UserInfo;
+import com.optimove.android.main.tools.networking.HttpClient;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import okhttp3.Response;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class OptimovePreferenceCenter {
+    private static final String TAG = "OptimovePrefCenter";
+    private static OptimovePreferenceCenter shared;
+
+    static ExecutorService executorService;
+
+    static final Handler handler = new Handler(Looper.getMainLooper());
+
+    public interface PreferencesGetHandler {
+        void run(ResultType result, @Nullable Preferences preferences);
+    }
+
+    public interface PreferencesSetHandler {
+        void run(ResultType result);
+    }
+
+    public enum ResultType {
+        SUCCESS,
+        ERROR_USER_NOT_SET,
+        ERROR
+    }
+
+    public enum Channel {
+        MOBILE_PUSH(489), WEB_PUSH(490), SMS(493), IN_APP(427), WHATSAPP(498), MAIL(15), INBOX(495);
+        private final int channel;
+
+        Channel(int channel) {
+            this.channel = channel;
+        }
+
+        @NonNull
+        public int getValue() {
+            return channel;
+        }
+
+        private static Channel getChannelByValue(int value) {
+            switch (value) {
+                case 489:
+                    return MOBILE_PUSH;
+                case 490:
+                    return WEB_PUSH;
+                case 493:
+                    return SMS;
+                case 427:
+                    return IN_APP;
+                case 498:
+                    return WHATSAPP;
+                case 15:
+                    return MAIL;
+                case 495:
+                    return INBOX;
+                default:
+                    throw new IllegalArgumentException("Preference center does not support channel " + value);
+            }
+        }
+    }
+
+    public static OptimovePreferenceCenter getInstance() {
+        if (shared == null) {
+            throw new IllegalStateException("OptimovePreferenceCenter is not initialized");
+        }
+        return shared;
+    }
+
+    /**
+     * Initializes an instance of OptimovePreferenceCenter
+     * <p>
+     * This API is intended for internal SDK use. Do not call this API or depend on it in your app.
+     */
+    public static void initialize() {
+        shared = new OptimovePreferenceCenter();
+        executorService = Executors.newSingleThreadExecutor();
+    }
+
+    /**
+     * Asynchronously runs preferences get handler on UI thread. Handler receives a single argument Preferences
+     *
+     * @param preferencesGetHandler handler
+     */
+    public void getPreferencesAsync(@NonNull PreferencesGetHandler preferencesGetHandler) {
```

Comment:
what's the behaviour now if they try to call GET/PUT before PC credentials were provided?

Reply:
Just pushed a change as it was failing getting region with config being null, now when config is null it will fire callback with ERROR result